File: targets/python-web/vulns/vuln02_sqli_login.py
import hashlib
from flask import Blueprint, request, jsonify
from store import get_db
bp = Blueprint("vuln02", __name__)

@bp.post("/vuln02/login")
def login():
    u = request.form.get("username", "")
    p = request.form.get("password", "")
    h = hashlib.md5(p.encode()).hexdigest()
    con = get_db()
    # VULN:VULN-02:CWE-89:taint SQL injection in authentication (login bypass)
    row = con.execute("SELECT username, role FROM users WHERE username = '" + u + "' AND password = '" + h + "'").fetchone()
    return jsonify({"status": "ok", "user": row["username"]} if row else {"status": "invalid"})

File: targets/jsts-web/src/vulns/vuln02_sqli_login.ts
import { Router } from "express";
import * as crypto from "crypto";
import { db } from "../store";
export const router = Router();
router.post("/vuln02/login", (req, res) => {
  const u = String(req.body.username || "");
  const h = crypto.createHash("md5").update(String(req.body.password || "")).digest("hex");
  // VULN:VULN-02:CWE-89:taint SQL injection in authentication (login bypass)
  const row: any = db.prepare("SELECT username, role FROM users WHERE username = '" + u + "' AND password = '" + h + "'").get();
  res.json(row ? { status: "ok", user: row.username } : { status: "invalid" });
});
